Nodejs/Product/Nodejs/NpmUI/NpmPackageInstallViewModel.cs

```diff
@@ -357,17 +367,64 @@ public PackageCatalogEntryViewModel SelectedPackage
             }
         }
 
+        public bool IsExecutingCommand
+        {
+            get
+            {
+                return this.isExecutingCommand;
+            }
+            set
+            {
+                this.isExecutingCommand = value;
+
+                OnPropertyChanged();
+                OnPropertyChanged("ExecutionProgressVisibility");
+            }
+        }
+
+        public Visibility ExecutionProgressVisibility
+        {
+            get { return IsExecutingCommand ? Visibility.Visible : Visibility.Collapsed; }
+        }
+
         #endregion
 
         public event PropertyChangedEventHandler PropertyChanged;
 
         protected virtual void OnPropertyChanged([CallerMemberName] string propertyName = null)
         {
-            var handler = this.PropertyChanged;
-            if (handler != null)
+            this.PropertyChanged?.Invoke(this, new PropertyChangedEventArgs(propertyName));
+        }
+
+        protected virtual void Dispose(bool disposing)
+        {
+            if (!disposed)
             {
-                handler(this, new PropertyChangedEventArgs(propertyName));
+                if (disposing)
+                {
+                    this.filterTimer?.Dispose();
+                }
+
+                var controller = this.npmController;
+                if (controller != null)
+                {
+                    this.npmController.FinishedRefresh -= this.NpmController_FinishedRefresh;
+                }
+                var worker = this.npmWorker;
+                if (worker != null)
+                {
+                    this.npmWorker.CommandStarted -= this.NpmWorker_CommandStarted;
+                    this.npmWorker.CommandCompleted -= this.NpmWorker_CommandCompleted;
+                }
+                disposed = true;
             }
         }
+
+        // This code added to correctly implement the disposable pattern.
+        public void Dispose()
+        {
+            // Do not change this code. Put cleanup code in Dispose(bool disposing) above.
+            Dispose(true);
```

Comment:
`GC.SuppressFinalize(this);`?

Reply:
only required when you override the finalizer